server/reports/attendance/attendance-endpoint.js

```diff
@@ -67,47 +61,47 @@ export function attendanceRequestHandler(request, response) {
-      return MindbodyAccess.getClasses();
+      return MindbodyAccess.getClasses({
+        StartDateTime: startdate,
+        EndDateTime: enddate,
+      });
     })
     // gets all classes from "Classes" endpoint
     .then((classes) => {
       let attendanceReport = {
         data: [],
-        headers: ["classId", "class", "capacity", "registered", "attended"],
+        headers: ["classId", "classTitle", "classStartTime", "capacity", "registered", "attended"],
       };
       let allNumberAttendedPromises = [];
       let numberOfClasses = Object.keys(classes.Classes).length;
 
       // iterates through every class in classes
       for (let i = 0; i < numberOfClasses; ++i) {
         let classId = classes.Classes[i].Id;
-        let classDate = fixDateFormat(classes.Classes[i].StartDateTime); // Mindbody StartDateTime format is yyyy/mm/dd by default
 
-        // adds class data to attendance report if startdate <= classdate <= enddate
-        if (isValidDate(startdate, classDate, enddate)) {
-          // adds attendance parameter to classData
-          let numberAttendedPromise = getNumberAttended(classId)
-            .then((numberAttended) => {
-              attendanceReport.data.push([
-                classId,
-                classes.Classes[i].ClassDescription.Name,
-                classes.Classes[i].MaxCapacity, 
-                classes.Classes[i].TotalBooked, 
-                numberAttended
-              ]); // pushes current class's data to attendanceReport
-            })
-            .catch((err) => {
-              console.log("ERROR! ", err);
-            });
-          allNumberAttendedPromises.push(numberAttendedPromise);
-        }
+        // adds attendance parameter to classData
+        let numberAttendedPromise = getNumberAttended(classId)
+          .then((numberAttended) => {
+            attendanceReport.data.push([
+              classId,
+              classes.Classes[i].ClassDescription.Name,
+              classes.Classes[i].StartDateTime,
+              classes.Classes[i].MaxCapacity, 
+              classes.Classes[i].TotalBooked, 
+              numberAttended
+            ]); // pushes current class's data to attendanceReport
+          })
+          .catch((err) => {
+            console.log("ERROR! ", err);
+          });
+        allNumberAttendedPromises.push(numberAttendedPromise);
       }
 
       // resolves all, allNumberAttendedPromises then outputs attendance report to endpoint as CSV or JSON
       Promise.all(allNumberAttendedPromises)
         .then(() => {
           if (format === "csv") {
-            let fields = ["class", "capacity", "registered", "attended"];
+            let fields = ["classId", "classTitle", "classStartTime", "capacity", "registered", "attended"];
             let parser = new J2C.Parser({ fields });
             let csv = parser.parse(attendanceReport);
             let fileName = "AttendanceReport.csv";
```

Comment:
CSV download doesn't work for me and always gives me and empty report. Does it work for you? ll push one way to fix it in a attendance-fixes-and-changes_fix1 branch.

Reply:
I've never had that problem. I've always gotten back a csv with a full report.

Reply:
When you say "CSV download doesn't work for me and always gives me an empty report" do you mean the download never happens and nothing is output to the endpoint, or the download happens, but nothing is in the .csv file?

Reply:
Hmm... Sorry I just tried again and I got a CSV output with no data in it. I have a ton of CSV files that I got back with data in it, so I don't know at what point this stopped working. It looks like you were able to fix it in PR #50. Thank you for catching that.